src/monodroid/jni/monodroid-glue.c

```diff
@@ -3031,13 +3031,13 @@ _monodroid_get_display_dpi (float *x_dpi, float *y_dpi)
 }
 
 static void
-lookup_bridge_info (MonoImage *image, const MonoJavaGCBridgeType *type, MonoJavaGCBridgeInfo *info)
+lookup_bridge_info (MonoDomain *domain, MonoImage *image, const MonoJavaGCBridgeType *type, MonoJavaGCBridgeInfo *info)
 {
-	info->klass             = mono.mono_class_from_name (image, type->namespace,    type->typename);
-	info->handle            = mono.mono_class_get_field_from_name (info->klass,     "handle");
-	info->handle_type       = mono.mono_class_get_field_from_name (info->klass,     "handle_type");
-	info->refs_added        = mono.mono_class_get_field_from_name (info->klass,     "refs_added");
-	info->weak_handle       = mono.mono_class_get_field_from_name (info->klass,     "weak_handle");
+	info->klass             = monodroid_get_class_from_image (&mono, domain, image, type->namespace, type->typename);
```

Comment:
Very important question: is `MonoClass*` a per-AppDomain value? What gives me pause is that `MonoJavaGCBridgeInfo::klass` is used from the GC thread, and thus from *any* (*every*) AppDomain, and is used with `mono_class_is_subclass_of()`, among other functions. If `MonoClass*` is a per-AppDomain value, *everything* is wrong, and then it becomes *really* important to unify with [the `java-interop` logic][0].

[0]: https://github.com/xamarin/java.interop/blob/master/src/java-interop/java-interop-gc-bridge-mono.c